web/frontend/src/components/slam/index.svelte

```diff
@@ -64,8 +64,15 @@ const refresh2d = async () => {
      * to see if a change has been made to the pointcloud map.
      * A new call to getPointCloudMap is made if an update has occured.
      */
-    if (mapTimestamp?.getSeconds() === lastTimestamp.getSeconds()) {
-      nextPose = await getPosition($robotClient, name);
+
+    const seconds = mapTimestamp?.getSeconds()
+    const nanos = mapTimestamp?.getNanos()
+
+    if (seconds !== undefined && nanos !== undefined) {
+      if (seconds > lastTimestamp.getSeconds() ||
```

Comment:
1. won't this fail to get the position if the timestamp is the same as last time?
2. Given we are doing this comparison in a few places, could we please make it a function? something like:
```
const localizationMode = (mapTimestamp) => {
if (mapTimestamp == undefined) {
return false
}
    const seconds = mapTimestamp.getSeconds()
    const nanos = mapTimestamp.getNanos()
    return seconds === lastTimestamp.getSeconds() && nanos === lastTimestamp.getNanos()
}
```

so that here we can do:
```
if (localizationMode(mapTimestamp)) {
nextPose = await getPosition($robotClient, name);
} else {
...
 ```